modules/claims_api/app/sidekiq/claims_api/reporting_base.rb

```diff
@@ -0,0 +1,140 @@
+# frozen_string_literal: true
+
+require 'claims_api/cid_mapper'
+
+module ClaimsApi
+  class ReportingBase < ClaimsApi::ServiceBase
+    def unsuccessful_claims_submissions
+      errored_claims.pluck(:cid, :created_at, :id).map do |cid, created_at, id|
+        { id:, created_at:, cid: }
+      end
+    end
+
+    def errored_claims
+      ClaimsApi::AutoEstablishedClaim.where(
+        created_at: @from..@to,
+        status: %w[errored]
+      ).order(:cid, :status)
+    end
+
+    def with_special_issues(cid: nil)
+      claims = ClaimsApi::AutoEstablishedClaim.where(created_at: @from..@to)
+      claims = claims.where(cid:) if cid.present?
+
+      claims.map { |claim| claim[:special_issues].length.positive? ? 1 : 0 }.sum.to_f
+    end
+
+    def with_flashes(cid: nil)
+      claims = ClaimsApi::AutoEstablishedClaim.where(created_at: @from..@to)
+      claims = claims.where(cid:) if cid.present?
+
+      claims.map { |claim| claim[:flashes].length.positive? ? 1 : 0 }.sum.to_f
+    end
+
+    def claims_totals
+      @claims_consumers.map do |cid|
+        counts = ClaimsApi::AutoEstablishedClaim.where(created_at: @from..@to, cid:).group(:status).count
+        totals = counts.sum { |_k, v| v }.to_f
+
+        percentage_with_flashes = "#{((with_flashes(cid:) / totals) * 100).round(2)}%"
+        percentage_with_special_issues = "#{((with_special_issues(cid:) / totals) * 100).round(2)}%"
+
+        if totals.positive?
+          consumer_name = ClaimsApi::CidMapper.new(cid:).name
+          {
+            consumer_name => counts.merge(totals:,
+                                          percentage_with_flashes: percentage_with_flashes.to_s,
+                                          percentage_with_special_issues: percentage_with_special_issues.to_s)
+                                   .deep_symbolize_keys
+          }
+        end
+      end
+    end
+
+    def monthly_claims_totals
```

Comment:
this is the only new method added in here.  The rest of the additions are a refactor because they are shared between the monthly and nightly job now (mentioned above)